.circleci/config.yml

```diff
@@ -168,8 +150,301 @@ jobs:
           command: |
             echo "Dummy job."
 
+  go-lint-test-build:
+    parameters:
+      binary_name:
+        description: Binary name to build
+        type: string
+      working_directory:
+        description: Working directory
+        type: string
+    docker:
+      - image: mslipper/go-builder:latest
+      - image: cimg/postgres:14.1
+    steps:
+      - checkout
+      - run:
+          name: Lint
+          command: golangci-lint run -E goimports -E sqlclosecheck -E bodyclose -E asciicheck ./...
+          working_directory: <<parameters.working_directory>>
+      - run:
+          name: Test
+          command: |
+            mkdir -p /test-results
+            gotestsum --junitfile /test-results/tests.xml
+          working_directory: <<parameters.working_directory>>
+      - store_test_results:
+          path: /test-results
+      - run:
+          name: Build
+          command: make <<parameters.binary_name>>
+          working_directory: <<parameters.working_directory>>
+
+  yarn-monorepo:
+    docker:
+      - image: mslipper/slither:latest
+    steps:
+      - restore_cache:
+          keys:
+            - v1-source-{{ .Branch }}-{{ .Revision }}
+            - v1-source-{{ .Branch }}
+      - checkout
+      - save_cache:
+          key: v1-source-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - ".git"
+      - restore_cache:
+          keys:
+            - v1-yarn-install-{{ checksum "yarn.lock" }}
+            - v1-yarn-install
+      - run:
+          name: Install dependencies
+          command: yarn --frozen-lockfile
+      - save_cache:
+          key: v1-yarn-install-{{ checksum "yarn.lock" }}
+          paths: 
+            - node_modules
+            - packages/common-ts/node_modules
+            - packages/contracts/node_modules
+            - packages/core-utils/node_modules
+            - packages/data-transport-layer/node_modules
+            - packages/message-relayer/node_modules
+            - packages/replica-healthcheck/node_modules
+            - packages/sdk/node_modules
+            - integration-tests/node_modules
+      - run:
+          name: Build monorepo
+          command: yarn build
+      - save_cache:
+          key: v1-yarn-build-{{ .Revision }}
+          paths:
+            - "."
+
+  contracts-slither:
+    docker:
+      - image: mslipper/slither:latest
+    steps:
+      - restore_cache:
+          keys:
+            - v1-yarn-build-{{ .Revision }}
+      - checkout
+      - run:
+          name: Run Slither
+          command: yarn test:slither
+          working_directory: packages/contracts
+
+  contracts-tests:
+    docker:
+      - image: mslipper/slither:latest
+    resource_class: xlarge
+    steps:
+      - restore_cache:
+          keys:
+            - v1-yarn-build-{{ .Revision }}
+      - checkout
+      - run:
+          name: Temporarily rename TS files
+          command: |
```

Comment:
Why does this step exist?

Reply:
The linter picks them up, which causes the linter to fail.

Reply:
They are automatically generated during compilation, so I can't add ignore comments to them unless we commit them.

Reply:
I can see if I can explicitly ignore those files in the `.eslintrc`.

Reply:
I could modify the auto-generation to add linter ignore comments at the top if `.eslintrc` isn't an option.